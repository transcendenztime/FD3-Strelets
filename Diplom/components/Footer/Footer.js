"use strict";

import React from 'react';

import PagesLinks from '../../pages/PagesLinks';
import CategoriesLinks from '../../pages/CategoriesLinks';

import '../../styles/Footer/Footer.css';

class Footer extends React.PureComponent {


  render() {

    return (
        <footer className = "footer">
          <div className = "footer_shadow"></div>
          <div className = "footer-nav">
            
            <div className="footer-nav__block footer-nav__block2">
              <h3 className="footer-nav__title">Информация</h3>
              <PagesLinks />
            </div>
            <div className="footer-nav__block footer-nav__block1">
              <h3 className="footer-nav__title">Категории</h3>
              <CategoriesLinks />
            </div>
            <div className="footer-nav__block footer-nav__block3">
              <h3 className="footer-nav__title">Контакты</h3>
              <p>г. Минск, ул. Змитрока Бядули, 15</p>
              <p>+375 (33) 325-50-11</p>
              <p>Время работы 09:00-18:00 (Пн.-Пт.)</p>
            </div>
          </div>
          <div className="footer-copyright">
            2003 - 2019 Долина растений - садовый интернет-магазин саженцев, семян, луковиц и деревьев.
          </div>
        </footer>
    );

  }

}

export default Footer;