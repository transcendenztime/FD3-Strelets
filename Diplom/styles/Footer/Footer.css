footer {
    position: relative;
    margin-top: 30px;
}
footer .footer_shadow {
    position: absolute;
    left: 0;
    right: 0;
    top: 0;
    height: 40px;
    background: #fff;
    background: -webkit-linear-gradient(top, #f2f2f2 0%, #fff 100%);
    background: -webkit-gradient(linear, left top, left bottom, color-stop(0%, #f2f2f2), color-stop(100%, #fff));
    background: -moz-linear-gradient(top, #f2f2f2 0%, #fff 100%);
    background: -ms-linear-gradient(top, #f2f2f2 0%, #fff 100%);
    background: -o-linear-gradient(top, #f2f2f2 0%, #fff 100%);
    filter: progid:DXImageTransform.Microsoft.gradient(startColorstr='#f2f2f2', endColorstr='#ffffff',GradientType=0 );
    z-index: 0;
}

.footer-nav {
	max-width: 1170px;
	margin: auto;
	display: flex;
	justify-content: space-around;
}
.footer-nav__block {
	margin: 20px;
}
.footer-nav__title {
	color: black;
	padding-left: 60px;	
    font-size: 17px;
    text-transform: uppercase;
    font-weight: bold;
}

.footer-nav__block a {
	text-decoration: none;
	font-size: 15px;
	font-weight: bold;
    outline: none;
}

.footer-nav__block .main-menu li {
    padding: 3px 0;
    height: 20px;
	font-size: 15px;
    font-weight: bold;
    color: #0d897f;
}
.footer-nav__block .main-menu a:hover {
    animation: footer-navLink-hover .5s forwards;
}
@keyframes footer-navLink-hover {
    0% {
        color: #0d897f;
    }
    100% {
        color: gray;
    }
}

.footer-copyright {
    background: #fff;
    background: -webkit-linear-gradient(top, #f2f2f2 0%, #fff 100%);
    background: -webkit-gradient(linear, left top, left bottom, color-stop(0%, #f2f2f2), color-stop(100%, #fff));
    background: -moz-linear-gradient(top, #f2f2f2 0%, #fff 100%);
    background: -ms-linear-gradient(top, #f2f2f2 0%, #fff 100%);
    background: -o-linear-gradient(top, #f2f2f2 0%, #fff 100%);
    filter: progid:DXImageTransform.Microsoft.gradient(startColorstr='#f2f2f2', endColorstr='#ffffff',GradientType=0 );
	
	margin: auto;
	padding: 20px 0;
    text-align: center;
    font-size: 14px;
}

.footer-nav__block .PageLink {    
	color: #0d897f;
    text-decoration: none;
    height: 20px;
    line-height: 20px;
}

.footer-nav .main-menu .category-menu  {
    display: none;
}


@media (max-width: 860px) { 
    .footer-nav  {
        font-size: 14px;
        flex-wrap: wrap-reverse;
    }
    .footer .category-menu .main-menu {
        padding-inline-start: 0;
    }
    .footer-nav__block3 {
        width: 100%;
        padding: 20px 0 0;
        text-align: center;
        line-height: 16px;
    }
    .footer-nav__block3  .footer-nav__title {
        padding-left: 0;	
    }
        
    .footer-nav__title {
        padding-left: 40px;	
        font-size: 16px;
    }

    .footer-nav__block a {
        font-size: 14px;
        padding: 0;
    }

    .footer-nav__block .main-menu li {
        height: 20px;
        font-size: 14px;
    }
    .footer-copyright {
        height: 20px;
        font-size: 12px;

    }
}

@media (max-width: 668px) {
    .footer-nav__block1, .footer-nav__block2 {
        display: none;
    }
}

@media (max-width: 340px) {
    .footer-copyright {
        font-size: 11px;
    }
}
