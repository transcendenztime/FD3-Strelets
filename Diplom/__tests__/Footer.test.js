"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import Footer from '../components/Footer/Footer';

test('внешний вид Footer', () => {

  // создаём тестовую версию компонента
  const component = renderer.create(
    <Router>
      <Footer />
    </Router>
  );

  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

    
});
