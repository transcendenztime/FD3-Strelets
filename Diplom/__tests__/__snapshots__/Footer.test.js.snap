// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`внешний вид Footer 1`] = `
<footer
  className="footer"
>
  <div
    className="footer_shadow"
  />
  <div
    className="footer-nav"
  >
    <div
      className="footer-nav__block footer-nav__block2"
    >
      <h3
        className="footer-nav__title"
      >
        Информация
      </h3>
      <ul
        className="main-menu"
      >
        <li>
          <a
            aria-current={null}
            className="PageLink"
            href="/"
            onClick={[Function]}
            style={undefined}
          >
            Главная
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="PageLink catalogueLink"
            href="/catalogue"
            onClick={[Function]}
            style={undefined}
          >
            Каталог
          </a>
          <ul
            className="category-menu main-menu"
          >
            <li>
              <a
                aria-current={null}
                className="CategoryLink PageLink"
                href="/catalogue/semena-ovoshchey"
                onClick={[Function]}
                style={undefined}
              >
                Семена овощей
              </a>
            </li>
            <li>
              <a
                aria-current={null}
                className="CategoryLink PageLink"
                href="/catalogue/semena-cvetov"
                onClick={[Function]}
                style={undefined}
              >
                Семена цветов
              </a>
            </li>
            <li>
              <a
                aria-current={null}
                className="CategoryLink PageLink"
                href="/catalogue/sazhency-roz"
                onClick={[Function]}
                style={undefined}
              >
                Саженцы роз
              </a>
            </li>
            <li>
              <a
                aria-current={null}
                className="CategoryLink PageLink"
                href="/catalogue/lukovicy-i-klubnelukovicy-cvetov"
                onClick={[Function]}
                style={undefined}
              >
                Луковицы и клубнелуковицы цветов
              </a>
            </li>
            <li>
              <a
                aria-current={null}
                className="CategoryLink PageLink"
                href="/catalogue/plodovye-derevya-i-kustarniki"
                onClick={[Function]}
                style={undefined}
              >
                Плодовые деревья и кустарники
              </a>
            </li>
            <li>
              <a
                aria-current={null}
                className="CategoryLink PageLink"
                href="/catalogue/hvoynye-rasteniya"
                onClick={[Function]}
                style={undefined}
              >
                Хвойные растения
              </a>
            </li>
            <li>
              <a
                aria-current={null}
                className="CategoryLink PageLink"
                href="/catalogue/dekorativnye-kustarniki"
                onClick={[Function]}
                style={undefined}
              >
                Декоративные кустарники
              </a>
            </li>
            <li>
              <a
                aria-current={null}
                className="CategoryLink PageLink"
                href="/catalogue/rododendrony"
                onClick={[Function]}
                style={undefined}
              >
                Рододендроны
              </a>
            </li>
          </ul>
        </li>
        <li>
          <a
            aria-current={null}
            className="PageLink"
            href="/warranty"
            onClick={[Function]}
            style={undefined}
          >
            Гарантии
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="PageLink"
            href="/partnership"
            onClick={[Function]}
            style={undefined}
          >
            Сотрудничество
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="PageLink"
            href="/contacts"
            onClick={[Function]}
            style={undefined}
          >
            Контакты
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="PageLink"
            href="/cart"
            onClick={[Function]}
            style={undefined}
          >
            Корзина
          </a>
        </li>
      </ul>
    </div>
    <div
      className="footer-nav__block footer-nav__block1"
    >
      <h3
        className="footer-nav__title"
      >
        Категории
      </h3>
      <ul
        className="category-menu main-menu"
      >
        <li>
          <a
            aria-current={null}
            className="CategoryLink PageLink"
            href="/catalogue/semena-ovoshchey"
            onClick={[Function]}
            style={undefined}
          >
            Семена овощей
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="CategoryLink PageLink"
            href="/catalogue/semena-cvetov"
            onClick={[Function]}
            style={undefined}
          >
            Семена цветов
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="CategoryLink PageLink"
            href="/catalogue/sazhency-roz"
            onClick={[Function]}
            style={undefined}
          >
            Саженцы роз
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="CategoryLink PageLink"
            href="/catalogue/lukovicy-i-klubnelukovicy-cvetov"
            onClick={[Function]}
            style={undefined}
          >
            Луковицы и клубнелуковицы цветов
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="CategoryLink PageLink"
            href="/catalogue/plodovye-derevya-i-kustarniki"
            onClick={[Function]}
            style={undefined}
          >
            Плодовые деревья и кустарники
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="CategoryLink PageLink"
            href="/catalogue/hvoynye-rasteniya"
            onClick={[Function]}
            style={undefined}
          >
            Хвойные растения
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="CategoryLink PageLink"
            href="/catalogue/dekorativnye-kustarniki"
            onClick={[Function]}
            style={undefined}
          >
            Декоративные кустарники
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="CategoryLink PageLink"
            href="/catalogue/rododendrony"
            onClick={[Function]}
            style={undefined}
          >
            Рододендроны
          </a>
        </li>
      </ul>
    </div>
    <div
      className="footer-nav__block footer-nav__block3"
    >
      <h3
        className="footer-nav__title"
      >
        Контакты
      </h3>
      <p>
        г. Минск, ул. Змитрока Бядули, 15
      </p>
      <p>
        +375 (33) 325-50-11
      </p>
      <p>
        Время работы 09:00-18:00 (Пн.-Пт.)
      </p>
    </div>
  </div>
  <div
    className="footer-copyright"
  >
    2003 - 2019 Долина растений - садовый интернет-магазин саженцев, семян, луковиц и деревьев.
  </div>
</footer>
`;
